src/BlocktrailSDK.php

```diff
@@ -564,20 +614,155 @@ public function createNewWallet($options) {
 
         // send the public keys to the server to store them
         //  and the mnemonic, which is safe because it's useless without the password
-        $data = $this->_createNewWallet($identifier, $primaryPublicKey, $backupPublicKey, $primaryMnemonic, $checksum, $keyIndex);
+        $data = $this->storeNewWalletV1($options['identifier'], $primaryPublicKey, $backupPublicKey, $primaryMnemonic, $checksum, $options['key_index']);
         // received the blocktrail public keys
         $blocktrailPublicKeys = $data['blocktrail_public_keys'];
 
-        $wallet = new Wallet($this, $identifier, $primaryMnemonic, [$keyIndex => $primaryPublicKey], $backupPublicKey, $blocktrailPublicKeys, $keyIndex, $this->network, $this->testnet, $checksum);
+        $wallet = new WalletV1(
+            $this,
+            $options['identifier'],
+            $primaryMnemonic,
+            [$options['key_index'] => $primaryPublicKey],
+            $backupPublicKey,
+            $blocktrailPublicKeys,
+            $options['key_index'],
+            $this->network,
+            $this->testnet,
+            $checksum
+        );
 
         $wallet->unlock($options);
 
         // return wallet and backup mnemonic
         return [
             $wallet,
-            $primaryMnemonic,
-            $backupMnemonic,
-            $blocktrailPublicKeys
+            [
+                'primary_mnemonic' => $primaryMnemonic,
+                'backup_mnemonic' => $backupMnemonic,
+                'blocktrail_public_keys' => $blocktrailPublicKeys,
+            ],
+        ];
+    }
+
+    public static function randomBits($bits) {
+        return self::randomBytes($bits / 8);
+    }
+
+    public static function randomBytes($bytes) {
+        return mcrypt_create_iv($bytes, \MCRYPT_DEV_URANDOM);
```

Comment:
IIRC, bitcoin-php and phpecc don't have a requirement for mcrypt anymore, so we should probably aim to do the same.